### PR TITLE
Marketplace: Improve partner submission success page

### LINKED ISSUE #64033: Marketplace: Improve partner submission success page
We need to iterate on https://wordpress.com/marketplace/submission-success to match the new design.
Previous PR: https://github.com/Automattic/wp-calypso/pull/63680
Figma: SJ6I0rUUtHNdHMF2KMMJig-fi-456%3A10526

Before and after:
frame-4461

Acceptance criteria:
- [x] The icons for each action/section have been replaced as per the Figma
- [x] We are adding an extra section taking users to our Developer Page (developer.wordpress.com)
- [x] We’re also adding a small footer note for users to get in touch with our teams directly. This will be similar (same links) that we have in our [existing partner’s page form.](https://wordpress.com/partners/).
- [x] Let’s remove the “Back to WordPress.com” link, as currently, that link would take users to their WPCOM calypso.

### PR DESCRIPTION
**Pull request by lupus2k**
#### Proposed Changes

We need to iterate on https://wordpress.com/marketplace/submission-success to match the new design.
Figma: SJ6I0rUUtHNdHMF2KMMJig-fi-456%3A10526

#### Testing Instructions

1. Go to `/marketplace/submission-success`
2. Check the design to match figma
3. Check `Developer Page` should redirect to `developer.wordpress.com`
4. Check the small footer and links should match links in our [existing partner’s page form.](https://wordpress.com/partners/)

##### Desktop
image

##### Mobile
image

Fixes #64033

### CODE REVIEW

File: client/my-sites/marketplace/pages/submission-success/signup-success.tsx [hunk 1]
```diff
@@ -1,10 +1,12 @@
-import { Gridicon } from '@automattic/components';
 import { ThemeProvider, Global, css } from '@emotion/react';
 import { Button } from '@wordpress/components';
 import { useTranslate } from 'i18n-calypso';
+import checkCircleImage from 'calypso/assets/images/marketplace/check-circle.png';
+import settingsImage from 'calypso/assets/images/marketplace/settings.png';
+import shoppingCartImage from 'calypso/assets/images/marketplace/shopping-cart.png';
```
Comment (allilevine) [inline client/my-sites/marketplace/pages/submission-success/signup-success.tsx]: @lupus2k Should we use svgs for these images? I've seen more svgs than pngs in Calypso.

**Comment by Rolilink** [comment]
Screen Shot 2022-06-07 at 12 45 15 PM

Icons are looking weird on mobile, maybe collapsing the button below the text would be a good solution there

**Review by Rolilink — CHANGES_REQUESTED** [review]
Let's do something about the mobile view before merging this one

**Comment by lupus2k** [comment]
> Let's do something about the mobile view before merging this one

What do you think to put the continue button under the text or should we ask for design?

File: client/my-sites/marketplace/pages/submission-success/signup-success.tsx (hunk 1 above)
Comment (lupus2k) [reply client/my-sites/marketplace/pages/submission-success/signup-success.tsx]: Sounds good, I will do it

**Comment by Rolilink** [comment]
> What do you think to put the continue button under the text

Let's try to do this one, if it doesn't look good we can talk to design

**Comment by lupus2k** [comment]
> > What do you think to put the continue button under the text
> 
> Let's try to do this one, if it doesn't look good we can talk to design

I tried with some CSS but I think it's not looking good. 
image

Can @SaxonF or @vinimotaa help us with some design for mobile on this page?

File: client/my-sites/marketplace/pages/submission-success/signup-success.tsx (hunk 1 above)
Comment (lupus2k) [reply client/my-sites/marketplace/pages/submission-success/signup-success.tsx]: Done

**Comment by SaxonF** [comment]
@lupus2k not sure if this helps but the same pattern is used in the intent step of onboarding which should scale for mobile

**Comment by lsl** [comment]
The desktop design doesn't match so unlikely to be much you can borrow, might just need to replicate locally.

Screenshot 2022-06-08 at 12-49-31 WordPress com

**Comment by lupus2k** [comment]
I updated CSS copying some styles from the intent step of onboarding as @SaxonF suggested.

**Review by Rolilink — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7367437

Thank you @lupus2k for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7367437).

**Pull request merged**